web/html-forms.md

```diff
@@ -0,0 +1,221 @@
+```
+<form>
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name" size="40">
+</form>
+```
+
+4. Required Attribute: This attribute tells that the specific input must be filled before submitting the form.
+```
+<form>
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name" required>
+</form>
+```
+
+5. Maxlength Attribute: It specifies the maximum number of characters allowed in an input field.
+```
+<form>
+  <label for="contact">Contact:</label>
+  <input type="number" id="contact" name="contact" maxlength="10">
+</form>
+```
+
+6. Minlength Attribute: It specifies the minimum number of characters allowed in an input field.
+```
+<form>
+  <label for="contact">Contact:</label>
+  <input type="number" id="contact" name="contact" minlength="8">
+</form>
+```
+
+7. Multiple Attribute: It specifies that the user is allowed to enter more than one value in an input field.
+```
+<form>
+  <label for="docs">Select Document:</label>
+  <input type="file" id="docs" name="docs" multiple>
+</form>
+```
+
+8. Step Attribute: specifies the legal number intervals for an input field.
+```
+<form>
+  <label for="marks">Marks:</label>
+  <input type="number" id="marks" name="marks" step="3">
+</form>
+```
+
+#### Submit and Action for Non AJAX Form
+- *Submit* : When the user enters the details which are asked, the data or the information is passed for further processing. Input field of type submit is taken to submit the data.
+
+- *Action* : The action attribute defines the action to be performed when the form is submitted.
+The method attribute specifies the HTTP method to be used when submitting the form data.
+It can be of the following types:
+1. Get : In this the form is sent as a URL variable. This is the default method while submitting the data.
+2. Post : In this the form is sent as HTTP post transaction. It is preferred to use when the form contains personal details.
+
+Following code, the snippet shows the illustration of the submission of the form.
+```
+<form action="/submit_action.php" method="get">
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name">
+  <label for="address">Address:</label>
+  <input type="text" id="address" name="address">
+   <input type="submit" value="Submit">
+</form>
+```
+In the above example the form data is sent to a page on the server called "/submit_action.php". This page contains a server-side script that handles the form data and the method is GET which will send the data as a URL parameter.
+
+
+### Common Mistakes / Misconceptions
+
+- **Javascript must be known to make an HTML form**
+Only basic knowledge of HTML is required to create a form in HTML.
+
+### Independent Practice
+Try making out forms and apply the concepts which you have learned from this. [Click here to practice](https://www.w3schools.com/html/exercise.asp?filename=exercise_html_forms2) .
+
+## Guided Practice
```

Comment:
This outline looks great!  I have 2 comments:
1. The order should always be: Guided Practice,  Independent Practice, Check for Understanding Supplemental Materials.
2.  Be sure to follow the AC's, which describe what independent practice should be:
Independent practice
Learn about what happens when a form is submitted. Are there echo/debug servers that can print out what is submitted to the server?

Target location: /web/html-forms.md

3. I like your current  Independent Practice activity, though... could have it in a "Check for understanding" section instead?

Reply:
1. I didn't get what do you mean by AC's .
2. Do I have to add the content about  "what happens when a form is submitted. Are there echo/debug servers that can print out what is submitted to the server"?
3. I have added the independent practice activity in check for understanding..now should I search something else for independent practice?